### PR TITLE
Simplify Lattice folder.

### PR DESCRIPTION
**Pull request by ye-luo**
Uniform3DGridLayout is a wrapper on top of CrysticalLattice and a bit of other things. However, non of the features other than CrysticalLattice is used. For this reason, I got rid of Uniform3DGridLayout layer and expose CrysticalLattice class directly. I introduced Lattice/LRBreakupParameters.h to handle parameters for LR break up. It is not clear to me what is the best way to keep these parameters but it seems good to keep the CrysticalLattice not getting over complicated.

### CODE REVIEW

**Comment by rcclay** [comment]
Thanks for pinging me.  I'll pick through it carefully and hopefully give you some feedback this afternoon or tomorrow.

**Comment by anbenali** [comment]
Heh!! I like this PR!!

File: src/Lattice/LRBreakupParameters.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
```
Comment (PDoakORNL) [inline src/Lattice/LRBreakupParameters.h]: src/Lattice/tests/test_LRBreakupParameters.cpp

File: src/Lattice/tests/test_bconds.cpp [hunk 2]
```diff
@@ -96,13 +96,13 @@ TEST_CASE("periodic_bulk_bconds", "[lattice]")
...
 {
-  Uniform3DGridLayout grid;
-  grid.BoxBConds = true; // periodic
+  CrystalLattice<OHMMS_PRECISION, OHMMS_DIM> Lattice;
+  Lattice.BoxBConds = true; // periodic
 
-  grid.R.diagonal(1.0);
-  grid.reset();
+  Lattice.R.diagonal(1.0);
+  Lattice.reset();
```
Comment (PDoakORNL) [inline src/Lattice/tests/test_bconds.cpp]: There seem to be no requirements for this test.

File: src/Lattice/CrystalLattice.h [hunk 3]
```diff
@@ -68,8 +68,8 @@ struct PosUnit
  *expression template operations with variable-cell algorithms.
  *
  */
-template<class T, unsigned D, bool ORTHO = false>
-struct CrystalLattice
+template<class T, unsigned D>
+struct CrystalLattice: public LRBreakupParameters<T, D>
```
Comment (PDoakORNL) [inline src/Lattice/CrystalLattice.h]: The class comment seems out of date.  Please avoid changing code without changing the explanatory comments. The indices comment is cryptic.  It looks like this facility is being removed from the code.

File: src/Lattice/tests/test_bconds.cpp (hunk 2 above)
Comment (ye-luo) [reply src/Lattice/tests/test_bconds.cpp]: This test seems useless. Removed it.

File: src/Lattice/LRBreakupParameters.h (hunk 1 above)
Comment (ye-luo) [reply src/Lattice/LRBreakupParameters.h]: Added

File: src/Lattice/tests/test_bconds.cpp [hunk 4]
```diff
@@ -13,14 +13,14 @@
 #define CATCH_CONFIG_MAIN
 #include "catch.hpp"
 
+#include <stdio.h>
```
[the remark this answers is not in the document]
Comment (ye-luo) [reply src/Lattice/tests/test_bconds.cpp]: Rename it.

File: src/Lattice/CrystalLattice.h (hunk 3 above)
Comment (ye-luo) [reply src/Lattice/CrystalLattice.h]: I removed the indices comment.

File: src/Particle/tests/test_particle.cpp [hunk 5]
```diff
@@ -98,46 +98,46 @@ TEST_CASE("particle set lattice with vacuum", "[particle]")
...
 
-  grid.VacuumScale = 2.0;
-  grid.reset();
+  Lattice.VacuumScale = 2.0;
+  Lattice.reset();
 
   source.setName("electrons");
-  source.Lattice.copy(grid);
+  source.Lattice.copy(Lattice);
   source.createSK();
```
[the remark this answers is not in the document]
Comment (ye-luo) [reply src/Particle/tests/test_particle.cpp]: This test is intended to check ParticleSet::createSK. I have no clue what these numbers are exactly.

**Comment by prckent** [comment]
Please restore the basic test in test_ParticleBConds.cpp . It  is a simple explicit test that checks that the boundary conditions are working and is easy for newcomers to understand so is not useless.

**Comment by ye-luo** [comment]
Restored the removed test.

File: src/Lattice/tests/test_bconds.cpp (hunk 2 above)
Comment (ye-luo) [reply src/Lattice/tests/test_bconds.cpp]: I reverted the removal and added some checks.

**Comment by ye-luo** [comment]
@PDoakORNL I've addressed your requests. Okay to merge?
